packages/js/src/lib/noCodeEvents.ts

```diff
@@ -40,23 +43,39 @@ export const addPageUrlEventListeners = (): void => {
   window.addEventListener("load", checkPageUrl);
 };
 
-export function checkUrlMatch(url: string, pageUrlValue: string, pageUrlRule: MatchType): boolean {
+export function checkUrlMatch(
+  url: string,
+  pageUrlValue: string,
+  pageUrlRule: MatchType
+): Result<boolean, InvalidMatchTypeError> {
+  let result: boolean;
+  let error: Result<never, InvalidMatchTypeError>;
+
   switch (pageUrlRule) {
     case "exactMatch":
-      return url === pageUrlValue;
+      result = url === pageUrlValue;
     case "contains":
-      return url.includes(pageUrlValue);
+      result = url.includes(pageUrlValue);
     case "startsWith":
-      return url.startsWith(pageUrlValue);
+      result = url.startsWith(pageUrlValue);
     case "endsWith":
-      return url.endsWith(pageUrlValue);
+      result = url.endsWith(pageUrlValue);
     case "notMatch":
-      return url !== pageUrlValue;
+      result = url !== pageUrlValue;
     case "notContains":
-      return !url.includes(pageUrlValue);
+      result = !url.includes(pageUrlValue);
     default:
```

Comment:
case falls through to error case. please add `break;`
(was missing before, I know)